dataprep/clean/clean_email.py

```diff
@@ -0,0 +1,514 @@
+    "yahoo.co.in",
+    "yahoo.com.sg",
+    "yahoo.com.ph",
+    "163.com",
+    "yeah.net",
+    "126.com",
+    "21cn.com",
+    "aliyun.com",
+    "foxmail.com",
+    # French ISP domains
+    "hotmail.fr",
+    "live.fr",
+    "laposte.net",
+    "yahoo.fr",
+    "wanadoo.fr",
+    "orange.fr",
+    "gmx.fr",
+    "sfr.fr",
+    "neuf.fr",
+    "free.fr",
+    # German ISP domains
+    "gmx.de",
+    "hotmail.de",
+    "live.de",
+    "online.de",
+    "t-online.de",
+    "web.de",
+    "yahoo.de",
+    # Italian ISP domains
+    "libero.it",
+    "virgilio.it",
+    "hotmail.it",
+    "aol.it",
+    "tiscali.it",
+    "alice.it",
+    "live.it",
+    "yahoo.it",
+    "email.it",
+    "tin.it",
+    "poste.it",
+    "teletu.it",
+    # Russian ISP domains
+    "mail.ru",
+    "rambler.ru",
+    "yandex.ru",
+    "ya.ru",
+    "list.ru",
+    # Belgian ISP domains
+    "hotmail.be",
+    "live.be",
+    "skynet.be",
+    "voo.be",
+    "tvcablenet.be",
+    "telenet.be",
+    # rgentinian ISP domains
+    "hotmail.com.ar",
+    "live.com.ar",
+    "yahoo.com.ar",
+    "fibertel.com.ar",
+    "speedy.com.ar",
+    "arnet.com.ar",
+    # Domains used in Mexico
+    "yahoo.com.mx",
+    "live.com.mx",
+    "hotmail.es",
+    "hotmail.com.mx",
+    "prodigy.net.mx",
+    # Domains used in Canada
+    "yahoo.ca",
+    "hotmail.ca",
+    "bell.net",
+    "shaw.ca",
+    "sympatico.ca",
+    "rogers.com",
+    # Domains used in Brazil
+    "yahoo.com.br",
+    "hotmail.com.br",
+    "outlook.com.br",
+    "uol.com.br",
+    "bol.com.br",
+    "terra.com.br",
+    "ig.com.br",
+    "itelefonica.com.br",
+    "r7.com",
+    "zipmail.com.br",
+    "globo.com",
+    "globomail.com",
+    "oi.com.br",
+}
+
+NEARBYKEYS = {
+    "a": ["q", "w", "s", "x", "z"],
+    "b": ["v", "g", "h", "n"],
+    "c": ["x", "d", "f", "v"],
+    "d": ["s", "e", "r", "f", "c", "x"],
+    "e": ["w", "s", "d", "r"],
+    "f": ["d", "r", "t", "g", "v", "c"],
+    "g": ["f", "t", "y", "h", "b", "v"],
+    "h": ["g", "y", "u", "j", "n", "b"],
+    "i": ["u", "j", "k", "o"],
+    "j": ["h", "u", "i", "k", "n", "m"],
+    "k": ["j", "i", "o", "l", "m"],
+    "l": ["k", "o", "p"],
+    "m": ["n", "j", "k", "l"],
+    "n": ["b", "h", "j", "m"],
+    "o": ["i", "k", "l", "p"],
+    "p": ["o", "l"],
+    "q": ["w", "a", "s"],
+    "r": ["e", "d", "f", "t"],
+    "s": ["w", "e", "d", "x", "z", "a"],
+    "t": ["r", "f", "g", "y"],
+    "u": ["y", "h", "j", "i"],
+    "v": ["c", "f", "g", "v", "b"],
+    "w": ["q", "a", "s", "e"],
+    "x": ["z", "s", "d", "c"],
+    "y": ["t", "g", "h", "u"],
+    "z": ["a", "s", "x"],
+    " ": ["c", "v", "b", "n", "m"],
+}
+
+STATS = {"cleaned": 0, "null": 0, "overflow": 0, "bad_format": 0, "unknown": 0}
+
+
+def clean_email(
+    df: Union[pd.DataFrame, dd.DataFrame],
+    column: str,
+    split: bool = False,
+    inplace: bool = False,
+    pre_clean: bool = False,
+    fix_domain: bool = False,
+    errors: str = "coerce",
+) -> pd.DataFrame:
```

Comment:
can we add a `report` [parameter](https://github.com/sfu-db/dataprep/blob/53059f94b9608ef6162d66a70f51a87db8ebfd51/dataprep/clean/clean_lat_long.py#L118) now like in `clean_lat_long`?